File: gachisa-backend/src/main/java/com/gachisa/order/dto/SavedDeliveryAddressResponse.java
package com.gachisa.order.dto;

import com.gachisa.order.entity.SavedDeliveryAddress;

public record SavedDeliveryAddressResponse(
        Long id,
        String addressName,
        String recipientName,
        String recipientPhone,
        String zipCode,
        String address,
        String addressDetail,
        String deliveryRequest
) {
    public static SavedDeliveryAddressResponse from(SavedDeliveryAddress saved) {
        return new SavedDeliveryAddressResponse(saved.getId(), saved.getAddressName(),
                saved.getRecipientName(), saved.getRecipientPhone(), saved.getZipCode(),
                saved.getAddress(), saved.getAddressDetail(), saved.getDeliveryRequest());
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/order/dto/SavedDeliveryAddressResponse.kt
package com.gachisa.order.dto

import com.gachisa.order.entity.SavedDeliveryAddress

data class SavedDeliveryAddressResponse(
    val id: Long,
    val addressName: String,
    val recipientName: String,
    val recipientPhone: String,
    val zipCode: String,
    val address: String,
    val addressDetail: String,
    val deliveryRequest: String?,
) {
    companion object {
        fun from(saved: SavedDeliveryAddress): SavedDeliveryAddressResponse = SavedDeliveryAddressResponse(
            id = saved.id!!,
            addressName = saved.addressName,
            recipientName = saved.recipientName,
            recipientPhone = saved.recipientPhone,
            zipCode = saved.zipCode,
            address = saved.address,
            addressDetail = saved.addressDetail,
            deliveryRequest = saved.deliveryRequest,
        )
    }
}
